pinot-hadoop/src/main/java/org/apache/pinot/hadoop/job/SegmentPreprocessingJob.java

```diff
@@ -218,34 +431,20 @@ private void setMaxNumRecordsConfigIfSpecified(Job job) {
    * @return Input schema
    * @throws IOException exception when accessing to IO
    */
-  private static Schema getSchema(Path inputPathDir)
+  private Schema getSchema(Path inputPathDir)
       throws IOException {
     FileSystem fs = FileSystem.get(new Configuration());
     Schema avroSchema = null;
     for (FileStatus fileStatus : fs.listStatus(inputPathDir)) {
       if (fileStatus.isFile() && fileStatus.getPath().getName().endsWith(".avro")) {
```

Comment:
we do use file status in the for loop to get the schema

Reply:
But the schema is fetched from `_dataStreamReader`, which has already been initialized somewhere else instead of here. `fileStatus` isn't used here at all